src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -373,6 +373,77 @@ mongoc_client_pool_try_pop (mongoc_client_pool_t *pool)
    RETURN (client);
 }
 
+static bool
+_mongoc_connect_client_to_servers (mongoc_client_t *client)
+{
+   bool ok = true;
+   bson_error_t error = {0};
+   bson_t *ping_command = BCON_NEW ("ping", BCON_INT32 (1));
+   size_t num_servers = 0;
+   mongoc_server_description_t **descriptions =
+      mongoc_client_get_server_descriptions (client, &num_servers);
```

Comment:
I suggest sending a `ping` to one server before calling `mongoc_client_get_server_descriptions`.

`mongoc_client_get_server_descriptions` does not connect to a server. If server monitoring has not yet returned, I expect it will return the initial set of server descriptions from the URI.

Example: if the URI is `mongodb://localhost:27017,localhost:27018`, but there are three nodes in the replica set, `mongoc_client_get_server_descriptions` may only return the two servers in the URI (if monitoring has not yet occurred). Once a `hello` response is processed from the two known servers, the third server will be added to the topology description.

This is applied in the suggested C++ implementation.